File: typescript/my-first-package/schemas/myFirstModule/structures.ts
// Module: schemas.myFirstModule.structures
import * as typelayerCommonDocumentation from "typelayer-common";


export class MyFirstStructure {
    myBooleanField: boolean;
    myDoubleField: number;
    myIntegerField: number;
    myListField: Array<string>;
    myMapField: Map<string, string>;
    myNestedField: MyEnumStructure;
    myTextField: string;
    constructor(arg: {myBooleanField: boolean; myDoubleField: number; myIntegerField: number; myListField: Array<string>; myMapField: Map<string, string>; myNestedField: MyEnumStructure; myTextField: string;}) {
        Object.assign(this, arg);
    }
}

export class MyEnumInteger {
    myTextField: number;
    constructor(arg: {myTextField: number;}) {
        Object.assign(this, arg);
    }
}

export class MyEnumText {
    myTextField: string;
    constructor(arg: {myTextField: string;}) {
        Object.assign(this, arg);
    }
}

export type MyEnumStructure = MyEnumInteger| MyEnumText

export function defaultMyFirstStructure(): MyFirstStructure {
    let myBooleanFieldVal : boolean;
    let myDoubleFieldVal : number;
    let myIntegerFieldVal : number;
    let myListFieldVal : Array<string>;
    let myMapFieldVal : Map<string, string>;
    let myNestedFieldVal : MyEnumStructure;
    let myTextFieldVal : string;
    myBooleanFieldVal = false;
    myDoubleFieldVal = 0.0;
    myIntegerFieldVal = 0;
    myListFieldVal = [];
    myMapFieldVal = new Map([]);
    myNestedFieldVal = defaultMyEnumStructure();
    myTextFieldVal = "";
    return new MyFirstStructure({myBooleanField: myBooleanFieldVal, myDoubleField: myDoubleFieldVal, myIntegerField: myIntegerFieldVal, myListField: myListFieldVal, myMapField: myMapFieldVal, myNestedField: myNestedFieldVal, myTextField: myTextFieldVal});
}

export function describeMyFirstStructure(): typelayerCommonDocumentation.StructureInfo {
    let myBooleanFieldFieldInfo : typelayerCommonDocumentation.TypeInfo;
    let myDoubleFieldFieldInfo : typelayerCommonDocumentation.TypeInfo;
    let myFirstStructureConstructorInfo : typelayerCommonDocumentation.ConstructorInfo;
    let myFirstStructureMyBooleanFieldFieldInfo : typelayerCommonDocumentation.FieldDescription;
    let myFirstStructureMyDoubleFieldFieldInfo : typelayerCommonDocumentation.FieldDescription;
    let myFirstStructureMyIntegerFieldFieldInfo : typelayerCommonDocumentation.FieldDescription;
    let myFirstStructureMyListFieldFieldInfo : typelayerCommonDocumentation.FieldDescription;
    let myFirstStructureMyMapFieldFieldInfo : typelayerCommonDocumentation.FieldDescription;
    let myFirstStructureMyNestedFieldFieldInfo : typelayerCommonDocumentation.FieldDescription;
    let myFirstStructureMyTextFieldFieldInfo : typelayerCommonDocumentation.FieldDescription;
    let myIntegerFieldFieldInfo : typelayerCommonDocumentation.TypeInfo;
    let myListFieldFieldInfo : typelayerCommonDocumentation.TypeInfo;
    let myListFieldInnerFieldInfo : typelayerCommonDocumentation.TypeInfo;
    let myMapFieldFieldInfo : typelayerCommonDocumentation.TypeInfo;
    let myMapFieldInnerFieldInfo : typelayerCommonDocumentation.TypeInfo;
    let myNestedFieldFieldInfo : typelayerCommonDocumentation.TypeInfo;
    let myTextFieldFieldInfo : typelayerCommonDocumentation.TypeInfo;
    myBooleanFieldFieldInfo = new typelayerCommonDocumentation.TlBoolean({});
    myFirstStructureMyBooleanFieldFieldInfo = new typelayerCommonDocumentation.FieldDescription({name: "my_boolean_field", description: null, typeInfo: myBooleanFieldFieldInfo});
    myDoubleFieldFieldInfo = new typelayerCommonDocumentation.TlDouble({});
    myFirstStructureMyDoubleFieldFieldInfo = new typelayerCommonDocumentation.FieldDescription({name: "my_double_field", description: null, typeInfo: myDoubleFieldFieldInfo});
    myIntegerFieldFieldInfo = new typelayerCommonDocumentation.TlInteger({});
    myFirstStructureMyIntegerFieldFieldInfo = new typelayerCommonDocumentation.FieldDescription({name: "my_integer_field", description: null, typeInfo: myIntegerFieldFieldInfo});
    myListFieldInnerFieldInfo = new typelayerCommonDocumentation.TlText({});
    myListFieldFieldInfo = new typelayerCommonDocumentation.TlList({listInner: myListFieldInnerFieldInfo});
    myFirstStructureMyListFieldFieldInfo = new typelayerCommonDocumentation.FieldDescription({name: "my_list_field", description: null, typeInfo: myListFieldFieldInfo});
    myMapFieldInnerFieldInfo = new typelayerCommonDocumentation.TlText({});
    myMapFieldFieldInfo = new typelayerCommonDocumentation.TlMap({mapInner: myMapFieldInnerFieldInfo});
    myFirstStructureMyMapFieldFieldInfo = new typelayerCommonDocumentation.FieldDescription({name: "my_map_field", description: null, typeInfo: myMapFieldFieldInfo});
    myNestedFieldFieldInfo = new typelayerCommonDocumentation.TlStructure({package: "my_first_package", schema: "structures", structureId: "my_enum_structure"});
    myFirstStructureMyNestedFieldFieldInfo = new typelayerCommonDocumentation.FieldDescription({name: "my_nested_field", description: null, typeInfo: myNestedFieldFieldInfo});
    myTextFieldFieldInfo = new typelayerCommonDocumentation.TlText({});
    myFirstStructureMyTextFieldFieldInfo = new typelayerCommonDocumentation.FieldDescription({name: "my_text_field", description: null, typeInfo: myTextFieldFieldInfo});
    myFirstStructureConstructorInfo = new typelayerCommonDocumentation.ConstructorInfo({name: "my_first_structure", description: null, fields: [myFirstStructureMyBooleanFieldFieldInfo, myFirstStructureMyDoubleFieldFieldInfo, myFirstStructureMyIntegerFieldFieldInfo, myFirstStructureMyListFieldFieldInfo, myFirstStructureMyMapFieldFieldInfo, myFirstStructureMyNestedFieldFieldInfo, myFirstStructureMyTextFieldFieldInfo]});
    return new typelayerCommonDocumentation.StructureInfo({structureId: "my_first_structure", structureType: new typelayerCommonDocumentation.StructureEnum({}), description: null, constructors: [myFirstStructureConstructorInfo]});
}

export function defaultMyEnumStructure(): MyEnumInteger {
    let myTextFieldVal : number;
    myTextFieldVal = 0;
    return new MyEnumInteger({myTextField: myTextFieldVal});
}

export function defaultMyEnumInteger(): MyEnumInteger {
    let myTextFieldVal : number;
    myTextFieldVal = 0;
    return new MyEnumInteger({myTextField: myTextFieldVal});
}

export function defaultMyEnumText(): MyEnumText {
    let myTextFieldVal : string;
    myTextFieldVal = "";
    return new MyEnumText({myTextField: myTextFieldVal});
}

export function describeMyEnumStructure(): typelayerCommonDocumentation.StructureInfo {
    let myEnumIntegerConstructorInfo : typelayerCommonDocumentation.ConstructorInfo;
    let myEnumIntegerMyTextFieldFieldInfo : typelayerCommonDocumentation.FieldDescription;
    let myEnumTextConstructorInfo : typelayerCommonDocumentation.ConstructorInfo;
    let myEnumTextMyTextFieldFieldInfo : typelayerCommonDocumentation.FieldDescription;
    let myTextFieldFieldInfo : typelayerCommonDocumentation.TypeInfo;
    myTextFieldFieldInfo = new typelayerCommonDocumentation.TlInteger({});
    myEnumIntegerMyTextFieldFieldInfo = new typelayerCommonDocumentation.FieldDescription({name: "my_text_field", description: null, typeInfo: myTextFieldFieldInfo});
    myEnumIntegerConstructorInfo = new typelayerCommonDocumentation.ConstructorInfo({name: "my_enum_integer", description: null, fields: [myEnumIntegerMyTextFieldFieldInfo]});
    myTextFieldFieldInfo = new typelayerCommonDocumentation.TlText({});
    myEnumTextMyTextFieldFieldInfo = new typelayerCommonDocumentation.FieldDescription({name: "my_text_field", description: null, typeInfo: myTextFieldFieldInfo});
    myEnumTextConstructorInfo = new typelayerCommonDocumentation.ConstructorInfo({name: "my_enum_text", description: null, fields: [myEnumTextMyTextFieldFieldInfo]});
    return new typelayerCommonDocumentation.StructureInfo({structureId: "my_enum_structure", structureType: new typelayerCommonDocumentation.StructureEnum({}), description: null, constructors: [myEnumIntegerConstructorInfo, myEnumTextConstructorInfo]});
}

File: python/my_first_package/my_first_package/schemas/my_first_module/structures.py
from typelayer_common import documentation as typelayer_common_documentation
from dataclasses import dataclass
from typing import List, Dict, Optional


@dataclass
class MyFirstStructure:
    my_boolean_field: bool
    my_double_field: float
    my_integer_field: int
    my_list_field: List[str]
    my_map_field: Dict[str, str]
    my_nested_field: "MyEnumStructure"
    my_text_field: str

class MyEnumStructure:
    pass


@dataclass
class MyEnumInteger(MyEnumStructure):
    my_text_field: int

@dataclass
class MyEnumText(MyEnumStructure):
    my_text_field: str

def default_my_first_structure() -> MyFirstStructure:
    
    
    
    
    
    
    
    my_boolean_field_val = False
    my_double_field_val = 0.0
    my_integer_field_val = 0
    my_list_field_val = []
    my_map_field_val = {}
    my_nested_field_val = default_my_enum_structure()
    my_text_field_val = ""
    return MyFirstStructure(my_boolean_field=my_boolean_field_val, my_double_field=my_double_field_val, my_integer_field=my_integer_field_val, my_list_field=my_list_field_val, my_map_field=my_map_field_val, my_nested_field=my_nested_field_val, my_text_field=my_text_field_val)

def describe_my_first_structure() -> typelayer_common_documentation.StructureInfo:
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    my_boolean_field_field_info = typelayer_common_documentation.TlBoolean()
    my_first_structure_my_boolean_field_field_info = typelayer_common_documentation.FieldDescription(name="my_boolean_field", description=None, type_info=my_boolean_field_field_info)
    my_double_field_field_info = typelayer_common_documentation.TlDouble()
    my_first_structure_my_double_field_field_info = typelayer_common_documentation.FieldDescription(name="my_double_field", description=None, type_info=my_double_field_field_info)
    my_integer_field_field_info = typelayer_common_documentation.TlInteger()
    my_first_structure_my_integer_field_field_info = typelayer_common_documentation.FieldDescription(name="my_integer_field", description=None, type_info=my_integer_field_field_info)
    my_list_field_inner_field_info = typelayer_common_documentation.TlText()
    my_list_field_field_info = typelayer_common_documentation.TlList(list_inner=my_list_field_inner_field_info)
    my_first_structure_my_list_field_field_info = typelayer_common_documentation.FieldDescription(name="my_list_field", description=None, type_info=my_list_field_field_info)
    my_map_field_inner_field_info = typelayer_common_documentation.TlText()
    my_map_field_field_info = typelayer_common_documentation.TlMap(map_inner=my_map_field_inner_field_info)
    my_first_structure_my_map_field_field_info = typelayer_common_documentation.FieldDescription(name="my_map_field", description=None, type_info=my_map_field_field_info)
    my_nested_field_field_info = typelayer_common_documentation.TlStructure(package="my_first_package", schema="structures", structure_id="my_enum_structure")
    my_first_structure_my_nested_field_field_info = typelayer_common_documentation.FieldDescription(name="my_nested_field", description=None, type_info=my_nested_field_field_info)
    my_text_field_field_info = typelayer_common_documentation.TlText()
    my_first_structure_my_text_field_field_info = typelayer_common_documentation.FieldDescription(name="my_text_field", description=None, type_info=my_text_field_field_info)
    my_first_structure_constructor_info = typelayer_common_documentation.ConstructorInfo(name="my_first_structure", description=None, fields=[my_first_structure_my_boolean_field_field_info, my_first_structure_my_double_field_field_info, my_first_structure_my_integer_field_field_info, my_first_structure_my_list_field_field_info, my_first_structure_my_map_field_field_info, my_first_structure_my_nested_field_field_info, my_first_structure_my_text_field_field_info])
    return typelayer_common_documentation.StructureInfo(structure_id="my_first_structure", structure_type=typelayer_common_documentation.StructureEnum(), description=None, constructors=[my_first_structure_constructor_info])

def default_my_enum_structure() -> MyEnumInteger:

    my_text_field_val = 0
    return MyEnumInteger(my_text_field=my_text_field_val)

def default_my_enum_integer() -> MyEnumInteger:

    my_text_field_val = 0
    return MyEnumInteger(my_text_field=my_text_field_val)

def default_my_enum_text() -> MyEnumText:

    my_text_field_val = ""
    return MyEnumText(my_text_field=my_text_field_val)

def describe_my_enum_structure() -> typelayer_common_documentation.StructureInfo:
    
    
    
    
    
    my_text_field_field_info = typelayer_common_documentation.TlInteger()
    my_enum_integer_my_text_field_field_info = typelayer_common_documentation.FieldDescription(name="my_text_field", description=None, type_info=my_text_field_field_info)
    my_enum_integer_constructor_info = typelayer_common_documentation.ConstructorInfo(name="my_enum_integer", description=None, fields=[my_enum_integer_my_text_field_field_info])
    my_text_field_field_info = typelayer_common_documentation.TlText()
    my_enum_text_my_text_field_field_info = typelayer_common_documentation.FieldDescription(name="my_text_field", description=None, type_info=my_text_field_field_info)
    my_enum_text_constructor_info = typelayer_common_documentation.ConstructorInfo(name="my_enum_text", description=None, fields=[my_enum_text_my_text_field_field_info])
    return typelayer_common_documentation.StructureInfo(structure_id="my_enum_structure", structure_type=typelayer_common_documentation.StructureEnum(), description=None, constructors=[my_enum_integer_constructor_info, my_enum_text_constructor_info])